Analysis/tasks.py

```diff
@@ -0,0 +1,152 @@
+import law
+import luigi
+import os
+import shutil
+import time
+import yaml
+from RunKit.sh_tools import sh_call
+from RunKit.checkRootFile import checkRootFileSafe
+
+from run_tools.law_customizations import Task, HTCondorWorkflow, copy_param, get_param_value
+from AnaProd.tasks import AnaTupleTask, DataMergeTask
+
+class HistProducerFileTask(Task, HTCondorWorkflow, law.LocalWorkflow):
+    max_runtime = copy_param(HTCondorWorkflow.max_runtime, 10.0)
+    n_cpus = copy_param(HTCondorWorkflow.n_cpus, 1)
+
+    def workflow_requires(self):
+        return { "anaTuple" : AnaTupleTask.req(self, branches=())}
+
+    def requires(self):
+        sample_name, prod_br = self.branch_data
+        if sample_name =='data':
+            return [ DataMergeTask.req(self,max_runtime=DataMergeTask.max_runtime._default, branches=())]
+        return [ AnaTupleTask.req(self, branch=prod_br, max_runtime=AnaTupleTask.max_runtime._default, branches=())]
+
+    def create_branch_map(self):
+        n = 0
+        branches = {}
+        anaProd_branch_map = AnaTupleTask.req(self, branch=-1, branches=()).create_branch_map()
+        sample_id_data = 0
+        for prod_br, (sample_id, sample_name, sample_type, input_file) in anaProd_branch_map.items():
+            if sample_type =='data' or 'QCD' in sample_name:
+                continue
+            branches[n] = (sample_name, prod_br)#, anaProd_output)
+            n+=1
+        branches[n+1] = ('data', 0)#, dataMerge_output)
+        return branches
+
+    def output(self):
+        sample_name, prod_br = self.branch_data
+        anaProd_output = AnaTupleTask.req(self, branch=prod_br).output().path
+        dataMerge_output = DataMergeTask.req(self).output().path
+        input_file = anaProd_output if sample_name != 'data' else dataMerge_output
+        inFile_idx_list = input_file.split('/')[-1].split('.')[0].split('_')
+        inFile_idx = f'_{inFile_idx_list[1]}' if len(inFile_idx_list)>1 else ''
+        if not os.path.isdir(self.central_Histograms_path()):
+            os.makedirs(self.central_Histograms_path())
+        hist_config = os.path.join(self.ana_path(), 'config', 'plot','histograms.yaml')
```

Comment:
Read yaml only once. define global var in this file similarly as we did for samples.yaml.

Reply:
ok